src/app/deeds/deeds.component.html

```diff
@@ -2,149 +2,15 @@
-            <mat-list-item class="deeds-page--contact--header">
-              <mat-icon>description</mat-icon>
-              Description:
-            </mat-list-item>
-            <mat-list-item id="deeds--page--deed--field">
-              {{deed.description}}
-            </mat-list-item>
-            <mat-divider></mat-divider>
-            <mat-list-item class="deeds-page--contact--header">
-              <mat-icon>place</mat-icon>
-              Location:
-            </mat-list-item>
-            <mat-list-item class="deeds--page--deed--field">
-              {{deed.location}}
-            </mat-list-item>
-            <mat-divider></mat-divider>
-            <mat-list-item class="deeds-page--contact--header">
-              <mat-icon>call</mat-icon>
-              Contact information:
-            </mat-list-item>
-            <mat-list-item class="deeds--page--deed--field">Name: {{deed.contact.name}}</mat-list-item>
-            <mat-list-item class="deeds--page--deed--field">Email: {{deed.contact.email}}</mat-list-item>
-            <mat-list-item class="deeds--page--deed--field">Phone number: {{deed.contact.phone}}</mat-list-item>
-
-            <mat-divider></mat-divider>
-
-            <mat-list-item class="deeds-page--contact--header">
-              <mat-icon>people</mat-icon>
-              Participants:
-            </mat-list-item>
-            <mat-list-item *ngFor="let user of deed.users"
-                           class="deeds--page--deed--field">
-              {{user.username}}
-            </mat-list-item>
-            <mat-list-item class="deeds-page--contact--header" *ngIf="deed.teamLeadId !== null">
-              <mat-icon>stars</mat-icon>
-              Team lead: &nbsp;
-              <div *ngFor="let user of deed.users">
-                <div *ngIf="user.id === deed.teamLeadId"
-                     [ngStyle]="{'color' : deed.teamLeadId === user.id ? '#3f51b5' : 'black'}">
-                  {{user.username}}
-                </div>
-              </div>
-            </mat-list-item>
-            <div class="deeds--page--participate--btn-wrapper">
-              <button mat-raised-button class="deeds--page--deed--btn" (click)="registerSolo(deed)"
-                      *ngIf="!deed.closed">Participate as solo
-              </button>
-              <button mat-raised-button class="deeds--page--deed--btn" (click)="openTeamRegistration(deed)"
-                      *ngIf="!deed.closed && deed.participation !== 'PARTICIPATE_AS_SOLO'">Participate as
-                team
-              </button>
-              <label *ngIf="deed.closed">Deed is closed for registration</label>
-            </div>
-          </mat-list>
-        </mat-expansion-panel>
-      </mat-accordion>
+    <mat-tab label="All good deeds">
+      <span *ngFor="let deed of deeds">
+        <span app-deed-model-display [deed]="deed" (registerSoloEvent)="registerSolo($event)" (registerTeamEvent)="openTeamRegistration($event)"></span>
```

Comment:
Could you explain what does this span do? Looks fishy

Reply:
ngFor conflicts with events ("Can't have multiple template bindings")